src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/AgoraDBStatus.scala

```diff
@@ -1,28 +1,37 @@
 package org.broadinstitute.dsde.agora.server.dataaccess
 
+import com.typesafe.scalalogging.StrictLogging
 import org.broadinstitute.dsde.agora.server.dataaccess.mongo.AgoraMongoClient._
 import org.broadinstitute.dsde.agora.server.dataaccess.permissions.PermissionsDataSource
 import org.broadinstitute.dsde.workbench.util.health.{HealthMonitor, SubsystemStatus}
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
-class AgoraDBStatus(dataSource: PermissionsDataSource) {
+class AgoraDBStatus(dataSource: PermissionsDataSource) extends StrictLogging {
 
   def mongoStatus()(implicit executionContext: ExecutionContext): Future[SubsystemStatus] = {
-    Future(getMongoDBStatus) map {
-      case Success(_) => HealthMonitor.OkStatus
-      case Failure(t) => HealthMonitor.failedStatus(t.getMessage)
-    }
+    for {
+      _ <- Future(logger.info("Retrieving Mongo status"))
```

Comment:
Since these are in futures, would it be helpful to generate and log a timestamp here too?

Reply:
Have an example? Right now the messages end up looking like this, logging the thread and time the log statements are executed:

```
[INFO] [10:51:47.641] [agora-akka.actor.default-dispatcher-3] o.b.d.a.s.dataaccess.AgoraDBStatus - Retrieving MySQL status
[INFO] [10:51:47.642] [agora-akka.actor.default-dispatcher-4] o.b.d.a.s.dataaccess.AgoraDBStatus - Retrieving Mongo status
[INFO] [10:51:47.643] [pool-2-thread-1] o.b.d.a.s.d.p.AgoraEntityPermissionsClient - Checking database version
[INFO] [10:51:48.329] [agora-akka.actor.default-dispatcher-4] o.b.d.a.s.dataaccess.AgoraDBStatus - Retrieved Mongo status: SubsystemStatus(true,None)
[INFO] [10:51:49.659] [agora-akka.actor.default-dispatcher-4] o.b.d.a.s.d.p.AgoraEntityPermissionsClient - Checked database version
[INFO] [10:51:49.843] [agora-akka.actor.default-dispatcher-4] o.b.d.a.s.dataaccess.AgoraDBStatus - Retrieved MySQL status: SubsystemStatus(true,None)
```

Reply:
I certainly could be wrong about this, but I would imagine that the timestamp here is whenever the Future gets around to executing the logger statement. If you generated a timestamp val just before the for block and add it to the log statement, it might be different.

Reply:
Hooray PR reviews. The above log messages are on the wrong threads. Akka Http uses the default dispatcher, while the status checks should always be on their own threads.

This is what the above **should** look like. I'll submit another PR after so I can do A/B testing.

```
[INFO] [11:00:59.313] [agora-dispatchers.health-monitor-dispatcher-10] o.b.d.a.s.dataaccess.AgoraDBStatus - Retrieving MySQL status
[INFO] [11:00:59.315] [agora-dispatchers.health-monitor-dispatcher-8] o.b.d.a.s.dataaccess.AgoraDBStatus - Retrieving Mongo status
[INFO] [11:00:59.316] [pool-1-thread-1] o.b.d.a.s.d.p.AgoraEntityPermissionsClient - Checking database version
[INFO] [11:00:59.544] [agora-dispatchers.health-monitor-dispatcher-10] o.b.d.a.s.dataaccess.AgoraDBStatus - Retrieved Mongo status: SubsystemStatus(true,None)
[INFO] [11:01:00.448] [agora-dispatchers.health-monitor-dispatcher-9] o.b.d.a.s.d.p.AgoraEntityPermissionsClient - Checked database version
[INFO] [11:01:00.635] [agora-dispatchers.health-monitor-dispatcher-10] o.b.d.a.s.dataaccess.AgoraDBStatus - Retrieved MySQL status: SubsystemStatus(true,None)
```

Reply:
> I would imagine that the timestamp here is whenever the Future gets around to executing the logger statement.

That's exactly what I'm trying to log. The status check `Future`s aren't completing within 1 minute, causing a status poll timeout. I now think it's because health monitor was running on the Akka Http thread pool, but I've been wrong many, many, many times this past week. Nth time's the charm... 🤞

Reply:
#297

Reply:
Gotcha, makes perfect sense!